API/Authentication.md

```diff
@@ -0,0 +1,95 @@
+# Authentication
+
+Authentication is the process of verifying someone's identity. There are many ways to verify identities over HTTP APIs, the most popular of which will be described below.
+
+When we talk about API authentication, it's important to distinguish its two components:
+1. the mechanism — the medium which the authentication protocol utilizes
+2. the method — the protocol for authenticating the user
+
+These two work in tandem and sometimes they're indistinguishable, but we'll try to explain each one separately and then show how they work together. Let's look at available mechanisms first.
+
+## Mechanisms
+
+As was mentioned above, a mechanism is a medium utilized by the authentication protocol. The architecture of HTTP allows for several authentication mechanisms.
+
+### Cookies
+
+Cookies are pieces of data exchanged between the client and the server through headers. The server sends one or more cookies via a `Set-Cookie` response header and then the browser saves them and includes them in subsequent requests to the server in a `Cookie` header. The browser can also send cookies to the server without receiving any first. 
+
+When setting cookies, the server can apply one or more restrictions, like expiration date and domains where they're supposed to be used. You can learn more about how cookies work [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies).
+
+There are two important cookie attributes you should know about:
+- `Secure` — When set, cookie will only be sent to the server and saved in the browser if the protocol through which they're sent/received is `https`. Insecure `http` protocol can't access cookies with that attribute. If you're communicating over `https`, make sure to set this attribute.
+- `HttpOnly` - These cookies cannot be accessed by JavaScript code. They are exchanged only between the browser and the server. If the cookie is used to store server-side session (a method described below), this attribute should be enabled because the cookie should be saved and used only by the browser, not read or changed by the user through JS. Note that this attribute doesn't prevent tampering of cookies as users can still edit them manually (for example, in a developer console).
+
```

Comment:
maybe mention that to the date, `Cookie + HttpOnly` is the safest option for storing sensitive information in a browser

Reply:
Did you mean `Secure + HttpOnly` @vr4b4c?

Reply:
done 👍